posts/inside-rust/2022-07-29-compiler-team-2022-midyear-report.md

```diff
@@ -0,0 +1,800 @@
+solution would be available within the next two years, so that's a total of 16
+out of 22 that expect some kind of solution within two years.
+
+  * Note: One respondent noted that the two year threshold embedded in the
+    survey was an "odd timeframe." I do not disagree with that remark. The
+    reality is that I was trying to find a happy medium between two extremes:
+    short timeframes don't allow enough time for significant work to be
+    accomplished, while excessively long timeframes (e.g. greater than five
+    years) are very difficult to predict realistically. But the two year
+    time frame was drawn somewhat out of a hat.
+
+Of the nineteen ambitions that reported some progress so far this year, fourteen
+of them (14/19) also said that at the start of the year, they had either *no
+goals* or *no plan* to achieve those goals. Ten of those fourteen say they now
+have a better understanding of the problem, and five of those fourteen say they
+now have a plan.
+
+  * Here is a question I want to put forward to our community: Do we, as a
+    project, need to put more energy into upfront planning? Or, given our
+    volunteer nature, will that just distract people from "the fun stuff" and
+    potentially burn out contributors? In other words: is this 5/14 "ambitions
+    that spent time making a concrete plan", is that a value we
+    should work on growing? Or should we just monitor it, and not actively try to
+    change it?
+
+  * Note: It is possible that my survey was itself somewhat flawed, so we need
+    to be careful in how we interpret this data. My intention was to try to
+    capture how well structured a given ambition's plan was by asking if it had
+    established milestones and/or a schedule. However, the survey results
+    include cases where the respondent did *not* indicate that they had made a
+    plan, but they *did* say that this year they did establish milestones.
+    Should I have interpreted that as implicitly saying that there *was* a plan
+    established? Or is the respondent saying that something significant was
+    accomplished despite the absence of a plan with concrete milestones?
+    Maybe we need to restructure future surveys to make these
+    "implied bounds" clearer (ha ha, that's a little Rust type system joke).
+
+One more thing I want to point out: five of the ambitions checked the box in the
+survey that said "some of our work has reached Rust programmers, but we do not
+know if it has improved Rust for them." (This is listed as "need-user-feedback"
+in the table above.) We should figure out ways to provide such feedback. I know
+that in one case, the lead for diagnostic improvements scours twitter to see if
+people are complaining about Rust's error messages (or celebrating them!).
+That's one approach, but I think we have to figure out something better. Perhaps
+we could provide an IDE plugin that provide telemetry to the project about the
```

Comment:
Given how controversial collecting telemetry is, we should at least throw "opt-in" here somewhere.

```suggestion
That's one approach, but I think we have to figure out something better. Perhaps
we could provide an opt-in plugin that provide telemetry to the project about the
```